packages/flame_3d/lib/src/resources/light/light.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:flame_3d/game.dart';
+import 'package:flame_3d/graphics.dart';
+import 'package:flame_3d/resources.dart';
+
+/// {@template light}
+/// A [Resource] that represents a light source that changes how the scene is
+/// rendered.
+///
+/// This class isn't a true resource, it does not upload it self to the GPU.
+/// Instead, it is used to modify how meshes are uploaded.
+/// {@endtemplate}
+class Light extends Resource<void> {
+  // TODO(luan): Add more attributes and light types
+  final Vector3 position;
+
+  /// {@macro light}
+  Light({
+    required this.position,
+  }): super(null);
```

Comment:
```suggestion
  /// {@macro light}
  Light({
    required this.position,
  }): super(null);
  
  final Vector3 position;
```

I know this isn't how we do it in Flame but that is how I did it in `flame_3d` :P so we should either keep it or refactor the rest.

Reply:
we can change that later to match Flame